mlflow/projects/databricks.py

```diff
@@ -27,6 +47,112 @@ def _jobs_runs_submit(req_body_json):
         endpoint="jobs/runs/submit", method="POST", req_body_json=req_body_json)
 
 
+def _get_databricks_run_cmd(dbfs_fuse_tar_uri, entry_point, parameters):
+    """
+    Generates MLflow CLI command to run on Databricks cluster in order to launch a run on Databricks
+    """
+    # Strip ".gz" and ".tar" file extensions from base filename of the tarfile
+    tar_hash = os.path.splitext(os.path.splitext(os.path.basename(dbfs_fuse_tar_uri))[0])[0]
+    container_tar_path = os.path.abspath(os.path.join(DB_TARFILE_BASE,
+                                                      os.path.basename(dbfs_fuse_tar_uri)))
+    project_dir = os.path.join(DB_PROJECTS_BASE, tar_hash)
+    mlflow_run_arr = list(map(shlex_quote, ["mlflow", "run", project_dir, "--new-dir",
+                                            "--entry-point", entry_point]))
+    if parameters is not None:
+        for key, value in parameters.items():
+            mlflow_run_arr.extend(["-P", "%s=%s" % (key, value)])
+    mlflow_run_cmd = " ".join(mlflow_run_arr)
+    shell_command = textwrap.dedent("""
+    export PATH=$PATH:$DB_HOME/python/bin:/$DB_HOME/conda/bin &&
+    mlflow --version &&
+    # Make local directories in the container into which to copy/extract the tarred project
+    mkdir -p {0} && mkdir -p {1} &&
+    # Rsync from DBFS FUSE to avoid copying archive into local filesystem if it already exists
+    rsync -a -v --ignore-existing {2} {0} &&
+    # Extract project into a temporary directory. We don't extract directly into the desired
+    # directory as tar extraction isn't guaranteed to be atomic
+    cd $(mktemp -d) &&
+    tar -xzvf {3} &&
+    # Atomically move the extracted project into the desired directory
+    mv -T {4} {5} &&
+    {6}
+    """.format(DB_TARFILE_BASE, DB_PROJECTS_BASE, dbfs_fuse_tar_uri, container_tar_path,
+               DB_TARFILE_ARCHIVE_NAME, project_dir, mlflow_run_cmd))
+    return ["bash", "-c", shell_command]
+
+
+def _check_databricks_cli_configured():
+    cfg_file = os.path.join(os.path.expanduser("~"), ".databrickscfg")
```

Comment:
Why do we check for the existence of this file? It seems like it would be better to leave the databricks CLI to figure out if it can find Databricks or not -- what if we add new or different paths in the future?